UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs

```diff
@@ -554,50 +561,65 @@ private void UpdateImageColor()
 		}
 	}
 
+	private void SetPaletteOnSpriteRenderer()
+	{
+		paletteSet = true;
+		var palette = getPaletteOrNull();
+		if (palette != null && palette.Count > 0 && palette.Count <= 256)
+		{
+			MaterialPropertyBlock block = new MaterialPropertyBlock();
+			spriteRenderer.GetPropertyBlock(block);
+			List<Vector4> pal = palette.ConvertAll<Vector4>((Color c) => new Vector4(c.r, c.g, c.b, c.a));
+			block.SetVectorArray("_ColorPalette", pal);
+			block.SetInt("_IsPaletted", 1);
+			block.SetInt("_PaletteSize", pal.Count);
+			spriteRenderer.SetPropertyBlock(block);
+		}
+		else
+		{
+			MaterialPropertyBlock block = new MaterialPropertyBlock();
+			spriteRenderer.GetPropertyBlock(block);
+			block.SetInt("_IsPaletted", 0);
+			spriteRenderer.SetPropertyBlock(block);
+		}
+	}
+
+	private void SetPaletteOnImage()
+	{
+		List<Color>paletteOrNull = getPaletteOrNull();
+
+		if (paletteOrNull != null && palette.Count > 0 && palette.Count <= 256)
+		{
+			List<Vector4> pal = paletteOrNull.ConvertAll((c) => new Vector4(c.r, c.g, c.b, c.a));
+			image.material.SetVectorArray("_ColorPalette", pal);
+			image.material.SetInt("_IsPaletted", 1);
+			image.material.SetInt("_PaletteSize", pal.Count);
+		}
+		else
+		{
+			image.material.SetInt("_IsPaletted", 0);
+		}
+	}
+
 	private void SetImageSprite(Sprite value)
 	{
-		List<Color> paletteOrNull;
 		if (spriteRenderer != null)
 		{
 			spriteRenderer.enabled = true;
 			spriteRenderer.sprite = value;
-			if (isPaletted() && PaletteSet == false)
-			{
-				PaletteSet = true;
-				var palette = getPaletteOrNull();
-				if (palette != null && palette.Count == 8)
-				{
-					MaterialPropertyBlock block = new MaterialPropertyBlock();
-					spriteRenderer.GetPropertyBlock(block);
-					List<Vector4> pal = palette.ConvertAll<Vector4>((Color c) => new Vector4(c.r, c.g, c.b, c.a));
-					block.SetVectorArray("_ColorPalette", pal);
-					block.SetInt("_IsPaletted", 1);
-					spriteRenderer.SetPropertyBlock(block);
-				}
-			}
-			else if (PaletteSet)
+
+			if (!paletteSet)
```

Comment:
I personally hate this one, but our code guidelines say to explicitly compare stuff to false (myVar == false) instead of using the ! operator.

Reply:
Ah, I might have some things to change in JavaScript land too. Thanks for the heads up

Reply:
Wow searching the source for `!\s*[\w(]` with regex mode gave me a lot of hits. I went ahead and made some extra files comply while I am at it.

Reply:
Yes, we have a lot of non compliant code but I try to keep new contributions in line with the code style.

Reply:
> I went ahead and made some extra files comply while I am at it

Be careful so we don't get conflicts with what other people are working on.